src/main/java/mate/academy/service/impl/MovieServiceImpl.java

```diff
@@ -19,11 +20,17 @@ public Movie add(Movie movie) {
 
     @Override
     public Movie get(Long id) {
-        return movieDao.get(id).get();
+        if (id == null || id < 0) {
+            throw new IllegalArgumentException("Invalid cinema hall ID: " + id);
+        }
+        return movieDao
+                .get(id)
+                .orElseThrow(() ->
+                new DataProcessingException("Movie service not found with ID:" + id));
```

Comment:
The exception message refers to 'Movie service not found with ID'. It would be more accurate to say 'Movie not found with ID' to reflect the entity being accessed.